aspnetcore/signalr/javascript-client.md

```diff
@@ -0,0 +1,108 @@
+
+  ```console
+   npm install @aspnet/signalr
+  ```
+
+Npm installs the package contents in the *node_modules\\@aspnet\signalr\dist\browser* folder. Copy the *signalr.js* or its minified version to a folder in your project.
+
+## Use the SignalR JavaScript client
+
+To use the SignalR JavaScript client, a script reference is required. Since the reference is an HTML `<script>` element, it can go in any kind of client-side page, such as a [Razor Page](xref:mvc/razor-pages/index?tabs=visual-studio), [MVC view](xref:mvc/views/overview), or HTML page.
+
+```html
+<script src="~/scripts/signalr.js"></script>
+```
+
+## Connect to a hub
+
+To connect to a SignalR hub, create a new instance of a connection with the route as its argument. The route maps to a hub with the same name. However, the hub's name can be mixed case, such as `ChatHub`.
+
+```javascript
+const connection = new signalR.HubConnection('/chathub');
+```
+
+While the preceding code sample creates a connection to a hub named `ChatHub`, it doesn't start it. To start the connection, call the `connection.start` method.
+
+```javascript
+connection.start().catch(err => logError(err));
+```
+
+### Specify the connection's transport type
+
+Transports define how servers and clients communicate. To set the transport type when you create the connection by setting the `signalR.TransportType` to *WebSockets*, *LongPolling*, or *ServerSentEvents*. The default, recommended transport is *WebSockets*.
+
+[!code-javascript[Specify transport](javascript-client/sample/chat.js?range=1-3)]
+
+### Cross-origin connections
+
+Typically, browsers load connections from the same domain as the requested page. However, there are occasions when a connection to another domain is required.
+
+For better security, [cross-origin connections](xref:security/cors) are disabled by default. If you need to create a cross-origin request, configure and enable it in the `Startup` class.
+
+[!code-csharp[Cross-origin connections](javascript-client/sample/startup.cs?highlight=22-29,39-40)]
+
+## Call hub methods from client
+
+JavaScript clients can call public methods on hubs by issuing `connection.invoke`. The `invoke` method accepts two arguments:
+
+* The name of the hub's method. In the following example, the hub name is `SendMessage`.
+* Any arguments defined in the hub's method. In the following example, the argument name is `message`.
+
+[!code-javascript[Call hub methods](javascript-client/sample/chat.js?range=13-16)]
+
+## Call client methods from hub
+
+To receive messages from the hub, define a method in the client's `connection.on` method. The `connection.on` method requires the following information:
+
+* The name of the JavaScript client's method. In the following example, the method name is `ReceiveMessage`.
+* Arguments the hub passes to the method. In the following example, the argument value is `message`.
+* The method's code.
```

Comment:
There are only two parameters to the `.on` method: The name of the method, and a JavaScript function to implement the method. I wouldn't describe the arguments and code as separate parameters, since they aren't parameters.

Reply:
Do you want it to list just the two actual arguments? I was hoping that saying "information" rather than "arguments" -- "The `connection.on` method requires the following information:" made it clear it wasn't just arguments and everything needed for the method.